baseplate/context/__init__.py

```diff
@@ -53,3 +53,7 @@ def __init__(self, name, context_factory):
     def on_server_span_created(self, context, server_span):
         context_attr = self.context_factory.make_object_for_context(self.name, server_span)
         setattr(context, self.name, context_attr)
+
+    def on_local_span_created(self, context, span, component_name):
+        context_attr = self.context_factory.make_object_for_context(self.name, span)
```

Comment:
Since we'd  be creating a new session when a local span is created ( https://github.com/reddit/baseplate/blob/master/baseplate/context/sqlalchemy.py#L101 ) and a new observer to go with it to close it, this shouldn't inadvertently close the server span's session I think?